monitor/api_call.go

```diff
@@ -0,0 +1,52 @@
+// Package monitor provides core and utility types and functions that are useful
+// to more than one of the modules that make up the CT monitor.
+package monitor
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-monitor/client"
+)
+
+// APICall contains the details of a call to one of the API endpoints of a CT
+// Log.
+type APICall struct {
+	Start    time.Time
+	End      time.Time
```

Comment:
As discussed in person, there's a good chance APICall.End will be needed to be explicitly stored so that all APICalls that occurred before a certain time can be obtained (as all events before a certain time translates to all APICalls that ended before a certain time).

However, as also discussed, it might be nice if this decision is left to the storage implementation - a Start and Duration can easily be converted into a Start and End.

Also as discussed, durations can be negative, so having a time.Duration here doesn't do anything to prevent silly things like API calls ending before they began, if, for example, something goes funny with the clock in between the two times being taken.

That being said, I want to think through whether there's anything outside of the storage implementation that might use this End value before making the change to Start and Duration.  Specifically, I suspect the availability calculating section of the monitor may need it, in which case I might prefer to keep it as Start and End.

In summary, I would like to keep this as Start and End for now, spend a small amount of time thinking about whether it is needed explicitly this way, and then, only if the answer to that is no, follow up with a PR to change End to Duration.

Reply:
As we discussed, both ways have their pros and cons, neither is obviously better, so that sounds good to me.